Makefile

```diff
@@ -291,3 +293,64 @@ generate-mocks: install-mockery
 .PHONY: echo-version
 echo-version:
 	@echo $(GIT_CLOSEST_TAG)
+
+.PHONY: proto
+proto:
+	# Generate gogo, gRPC-Gateway, swagger, go-validators output.
+	#
+	# -I declares import folders, in order of importance
+	# This is how proto resolves the protofile imports.
+	# It will check for the protofile relative to each of these
+	# folders and use the first one it finds.
+	#
+	# --gogo_out generates GoGo Protobuf output with gRPC plugin enabled.
+	# --grpc-gateway_out generates gRPC-Gateway output.
+	# --swagger_out generates an OpenAPI 2.0 specification for our gRPC-Gateway endpoints.
+	# --govalidators_out generates Go validation files for our messages types, if specified.
+	#
+	# The lines starting with Mgoogle/... are proto import replacements,
+	# which cause the generated file to import the specified packages
+	# instead of the go_package's declared by the imported protof files.
+	#
+	# $$GOPATH/src is the output directory. It is relative to the GOPATH/src directory
+	# since we've specified a go_package option relative to that directory.
+	#
+	# model/proto/jaeger.proto is the location of the protofile we use.
+	#
+	# TODO use Docker container instead of installed protoc
+	# (https://medium.com/@linchenon/generate-grpc-and-protobuf-libraries-with-containers-c15ba4e4f3ad)
+	#
+	protoc \
+		-I model/proto \
+		-I vendor/github.com/grpc-ecosystem/grpc-gateway/ \
+		-I vendor/github.com/gogo/googleapis/ \
+		-I vendor/ \
+		--gogo_out=plugins=grpc,\
+Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,\
+Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types,\
+Mgoogle/protobuf/empty.proto=github.com/gogo/protobuf/types,\
+Mgoogle/api/annotations.proto=github.com/gogo/googleapis/google/api:\
+$$GOPATH/src/github.com/jaegertracing/jaeger/model/ \
+		--grpc-gateway_out=\
+Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,\
+Mgoogle/protobuf/empty.proto=github.com/gogo/protobuf/types,\
+Mgoogle/api/annotations.proto=github.com/gogo/googleapis/google/api:\
+$$GOPATH/src/github.com/jaegertracing/jaeger/model \
+		--swagger_out=model/proto/openapi/ \
+		model/proto/jaeger.proto
+
+	# Workaround for https://github.com/grpc-ecosystem/grpc-gateway/issues/229.
+	sed -i '' "s/empty.Empty/types.Empty/g" model/jaeger.pb.gw.go
+
+	protoc \
+		-I model/proto \
+		--go_out=$$GOPATH/src/github.com/jaegertracing/jaeger/model/prototest/ \
+		model/proto/jaeger_test.proto
+
+proto-install:
+	go install \
+		./vendor/github.com/gogo/protobuf/protoc-gen-gogo \
+		./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway \
+		./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
```

Comment:
Get this on my Debian desktop when trying this command:

```
$ make proto-install
go install \
	./vendor/github.com/gogo/protobuf/protoc-gen-gogo \
	./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway \
	./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway/main.go:18:2: cannot find package "github.com/golang/glog" in any of:
	/home/ihier/proj/go/src/github.com/jaegertracing/jaeger/vendor/github.com/golang/glog (vendor tree)
	/usr/local/go/src/github.com/golang/glog (from $GOROOT)
	/home/ihier/proj/go/src/github.com/golang/glog (from $GOPATH)
make: *** [Makefile:352: proto-install] Error 1
```

Reply:
After finally getting it to work and generating Python code, I get this clearly invalid import:

```
from github.com.gogo.protobuf.gogoproto import gogo_pb2 as github_dot_com_dot_gogo_dot_protobuf_dot_gogop
roto_dot_gogo__pb2
```

Reply:
Same with C++:
```
#include "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options/annotations.pb.h"
#include "github.com/gogo/protobuf/gogoproto/gogo.pb.h"
```

Reply:
we may have to strip them post-generation. E.g. in python these two extra imports are not actually used.